### ISSUE TITLE
Python 3.5.3 compatibility

### ISSUE THREAD

**Issue by N-Coder** [N-Coder/studip-fuse#17]
```
  File "/home/finksim/src/studip-fuse/studip_fuse/studipfs/api/aiointerface.py", line 4, in <module>
    from typing import Any, AsyncContextManager, AsyncGenerator, Callable, Coroutine, Dict, Generic, NamedTuple, Optional, TypeVar, Union
ImportError: cannot import name 'AsyncContextManager'

  File "/home/finksim/src/studip-fuse/studip_fuse/studipfs/api/aiointerface.py", line 4, in <module>
    from typing import Any, AsyncGenerator, Callable, Coroutine, Dict, Generic, NamedTuple, Optional, TypeVar, Union
ImportError: cannot import name 'AsyncGenerator'

  File "/home/finksim/src/studip-fuse/studip_fuse/launcher/aioimpl/asyncio/aiohttp_client.py", line 115, in DownloadState
    EMPTY = enum.auto()
AttributeError: module 'enum' has no attribute 'auto'

  File "/home/finksim/src/studip-fuse/studip_fuse/studipfs/api/session.py", line 6, in <module>
    from typing import AsyncGenerator, List, Mapping, Tuple
ImportError: cannot import name 'AsyncGenerator'
```

[typing.AsyncGenerator](https://docs.python.org/3/library/typing.html#typing.AsyncGenerator) is new in version 3.5.4
[typing.AsyncContextManager](https://docs.python.org/3/library/typing.html#typing.AsyncContextManager) is new in version 3.6
[enum.auto](https://docs.python.org/3/library/enum.html#enum.auto) is new in version 3.6

**Comment by N-Coder** [comment]
should also check whether we could use an old aiohttp version under Ubuntu 16.04

**Comment by N-Coder** [comment]
`typing.AsyncContextManager` is backported by [typing_extensions](https://github.com/python/typing/blob/master/typing_extensions/README.rst), `typing.AsyncGenerator` is only exported by `typing_extensions` if related classes are also available, which is not the case on 3.5.3. It is available trough [trio-typing](https://github.com/python-trio/trio-typing) (which also requires trio and mypy).

`enum.auto()` can be simulated by using the [functional API](https://docs.python.org/3.5/library/enum.html#functional-api), using the [aenum package](https://pypi.org/project/aenum/) or simply defining the `int` values explicitly.

**Comment by ClusterJan** [comment]
Testing the new "Python3.5" branch results in a new error: 

```
Uncaught exception from FUSE operation readdir, returning errno.EINVAL: argument of type 'NotImplementedType' is not iterable
Traceback (most recent call last):
[...]
  File "/home/aschwege/.local/lib/python3.5/site-packages/cached_property.py", line 35, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/home/aschwege/.local/lib/python3.5/site-packages/studip_fuse/avfs/virtual_path.py", line 231, in content_options
    if field_name not in self.format_token_generators:
TypeError: argument of type 'NotImplementedType' is not iterable
```

**Comment by N-Coder** [comment]
[`__init_subclass__`](https://docs.python.org/3/reference/datamodel.html#object.__init_subclass__) used in [virtual_path.py:215](https://github.com/N-Coder/studip-fuse/blob/master/studip_fuse/avfs/virtual_path.py#L215) to initialize the `format_token_generators` was added in 3.6

**Comment by N-Coder** [comment]
should be fixed by d644f5742050599dccbb80a6741b7ff9d208d692 and 8558a4f72c1a45530559b6cf5dddbdea68344882
